llvm/docs/GitHub.rst

```diff
@@ -135,62 +135,73 @@ you won't encounter merge conflicts when landing the PR.
+  from your fork.
 
-If you observe an error message from the above informing you that your pull
-request is not mergeable, then that is likely because upstream has been
-modified since your pull request was authored in a way that now results in a
-merge conflict. You must first resolve this merge conflict in order to merge
-your pull request. In order to do that:
+* `Interactive rebase <https://git-scm.com/docs/git-rebase#_interactive_mode>`_
+  with fixups. This is the recommended method since you can control the final
+  commit message and check that the final commit looks as you expect. When
+  your local state is correct, remember to force-push to your branch and press
+  the merge button in GitHub's web interface afterwards.
 
-::
+* Merge using the GitHub command line interface. Switch to your branch locally
+  and run:
 
-  git fetch upstream
-  git rebase upstream/main
+  ::
 
-Then fix the source files causing merge conflicts and make sure to rebuild and
-retest the result. Then:
+    gh pr merge --squash --delete-branch
 
-::
+  If you observe an error message from the above informing you that your pull
+  request is not mergeable, then that is likely because upstream has been
+  modified since your pull request was authored in a way that now results in a
+  merge conflict. You must first resolve this merge conflict in order to merge
+  your pull request. In order to do that:
 
-  git add <files with resolved merge conflicts>
-  git rebase --continue
+  ::
 
-Finally, you'll need to force push to your branch one more time before you can
-merge:
+    git fetch upstream
+    git rebase upstream/main
 
-::
+  Then fix the source files causing merge conflicts and make sure to rebuild and
+  retest the result. Then:
 
-  git push -f
-  gh pr merge --squash --delete-branch
+  ::
+
+    git add <files with resolved merge conflicts>
+    git rebase --continue
+
+  Finally, you'll need to force push to your branch one more time before you can
+  merge:
+
+  ::
+
+    git push -f
+    gh pr merge --squash --delete-branch
 
-This force push may ask if you intend to push hundreds, or potentially
-thousands of patches (depending on how long it's been since your pull request
-was initially authored vs. when you intended to merge it). Since you're pushing
-to a branch in your fork, this is ok and expected. Github's UI for the pull
-request will understand that you're rebasing just your patches, and display
-this result correctly with a note that a force push did occur.
+  This force push may ask if you intend to push hundreds, or potentially
+  thousands of patches (depending on how long it's been since your pull request
+  was initially authored vs. when you intended to merge it). Since you're pushing
+  to a branch in your fork, this is ok and expected. Github's UI for the pull
+  request will understand that you're rebasing just your patches, and display
+  this result correctly with a note that a force push did occur.
```

Comment:
Maybe suggest `--force-with-lease` here, in case other people may have pushed to the branch in the meantime? If you are pushing lots of patches I would assume it's either being worked on by a lot of people or over a long period of time.

Reply:
This option is new to me, so I might be misunderstanding it.

In this context the branch you're rebasing onto is the upstream `main` so yes that has a lot of other authors. If you're the only author of the branch being rebased then there shouldn't be a problem. Unless you forgot that you had pushed a more recent version of the branch from a different machine (which I've done but luckily I had access to both so I could recover it).

If there were multiple authors on a single branch for a single PR, then you don't want to overwrite their changes.

If you have squashed your changes locally, you will want to replace them in the remote branch. So `--force` is what you'd want to use.

So a note recommending `--force-with-lease` for branches with multiple authors seems appropriate, what do you think? 

(and I will expand `-f` in the existing example to `--force` for clarity)

Reply:
I've added a note the first time we talk about force pushes.

I don't want to clutter every mention of --force with it, and I think there are some parts of the PR process where you actually do want to overwrite changes on the remote. So the minority of folks collaborating on a single PR will have to be aware of that.